### PR TITLE
vm: implement Array/Struct/Map opcodes

### LINKED ISSUE #879: Implement Neo 3 VM compound-type opcodes
Arrays, structs and maps related functionality: neo-project/neo-vm#241, neo-project/neo-vm#273.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #879 . Related #881 .

### CODE REVIEW

File: pkg/vm/vm_test.go [hunk 1]
```diff
@@ -2713,6 +2713,45 @@ func TestREMOVEMap(t *testing.T) {
 	assert.Equal(t, makeStackItem(false), vm.estack.Pop().value)
 }
 
+func testCLEARITEMS(t *testing.T, items ...StackItem) {
```
Comment (roman-khimov) [inline pkg/vm/vm_test.go:2782]: Why is it variadic?

File: pkg/vm/vm.go [hunk 2]
```diff
@@ -1492,10 +1496,24 @@ func cloneIfStruct(item StackItem) StackItem {
 	}
 }
 
-func makeArrayOfFalses(n int) []StackItem {
+func makeArrayOfType(n int, typ StackItemType) []StackItem {
 	items := make([]StackItem, n)
+	var c func() StackItem
```
Comment (roman-khimov) [inline pkg/vm/vm.go:1501]: I don't think this situation is worth creating this callback, you can check for `typ` validity once and then use a switch inside a range loop. Alternatively we also have a `Dup()` method that can be used.

Comment (fyrchik) [reply pkg/vm/vm.go]: Fixed.

File: pkg/vm/vm_test.go (hunk 1 above)
Comment (fyrchik) [reply pkg/vm/vm_test.go]: I have first come up with another set of tests.

Comment (fyrchik) [reply pkg/vm/vm_test.go]: Fixed.

**Pull request merged**